libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py

```diff
@@ -1154,6 +1161,7 @@ def similarity_search_with_score_by_vector(
             "embedding": embedding,
             "keyword_index": self.keyword_index_name,
             "query": remove_lucene_chars(kwargs["query"]),
+            "ef": self.effective_search_ratio,
```

Comment:
Should the effective_search_ratio be pased into `similarity_search_with_score_by_vector` instead of the `__init__`? I think this is more intuitive as you would also pass `k` here

Reply:
cc @jexp

I think it makes more sense to define it only once, but I'm fine either way

Reply:
I think we should have it in  `similarity_search_with_score_by_vector` method with a default of `1`. The reasons being:

- It aligns with how one would use `queryNodes`.
- Users can choose to modify `ef` in search time rather than initialization time
- Easier to maintain as we don't need to manage when certain parameters should be defined in `__init__` as opposed to method arguments.

IMO we can centralise these arguments in the `__init__` if it gets to the point where the method is overloaded with arguments

Reply:
Switched